src/app/ReadClient.cpp

```diff
@@ -435,30 +435,30 @@ void ReadClient::OnResponseTimeout(Messaging::ExchangeContext * apExchangeContex
     ShutdownInternal(CHIP_ERROR_TIMEOUT);
 }
 
-CHIP_ERROR ReadClient::ProcessAttributePath(AttributePathIB::Parser & aAttributePath, ClusterInfo & aClusterInfo)
+CHIP_ERROR ReadClient::ProcessAttributePath(AttributePathIB::Parser & aAttributePathParser,
+                                            ConcreteDataAttributePath & aAttributePath)
 {
-    CHIP_ERROR err = aAttributePath.GetNode(&(aClusterInfo.mNodeId));
-    if (err == CHIP_END_OF_TLV)
-    {
-        err = CHIP_NO_ERROR;
-    }
-    VerifyOrReturnError(err == CHIP_NO_ERROR, err = CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_PATH);
-
+    CHIP_ERROR err = CHIP_NO_ERROR;
     // The ReportData must contain a concrete attribute path
-    err = aAttributePath.GetEndpoint(&(aClusterInfo.mEndpointId));
+    err = aAttributePathParser.GetEndpoint(&(aAttributePath.mEndpointId));
     VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_PATH);
-    err = aAttributePath.GetCluster(&(aClusterInfo.mClusterId));
+    err = aAttributePathParser.GetCluster(&(aAttributePath.mClusterId));
     VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_PATH);
-    err = aAttributePath.GetAttribute(&(aClusterInfo.mAttributeId));
+    err = aAttributePathParser.GetAttribute(&(aAttributePath.mAttributeId));
     VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_PATH);
 
-    err = aAttributePath.GetListIndex(&(aClusterInfo.mListIndex));
+    DataModel::Nullable<ListIndex> listIndex;
+    err = aAttributePathParser.GetListIndex(&(listIndex));
     if (CHIP_END_OF_TLV == err)
     {
         err = CHIP_NO_ERROR;
     }
+    else if (listIndex.IsNull())
+    {
+        aAttributePath.mListOp = ConcreteDataAttributePath::ListOperation::AppendItem;
+    }
+    // TODO: Add ListOperation::ReplaceItem support. (Attribute path with valid list index)
```

Comment:
This still needs to error out in that case for now.  Otherwise we'll silently corrupt data if the server does in fact send a single item...